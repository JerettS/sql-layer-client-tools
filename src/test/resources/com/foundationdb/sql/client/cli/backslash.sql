CREATE TABLE t1(c1 int default 10, c2 varchar(32) character set utf8 collate en_ci);
CREATE TABLE t2(c1 decimal(10,5), c2 varchar(10) for bit data);
CREATE INDEX i1 ON t1(c1);
CREATE INDEX i2 ON t2(c1);
CREATE SEQUENCE s1;
CREATE SEQUENCE s2;
CREATE VIEW v1 AS SELECT * FROM t1;
CREATE VIEW v2 AS SELECT * FROM t2;
  \l test.foo
\l test.%
\l test.%1
\l test.t1
\l test.s1
\l test.v1
\li test.%.%
\li test.t1.%
\li test.t1.i1
\lq test.%
\lq test.s1
\lt test.%
\lt test.t1
\lv test.%
\lv test.v1
\d test.%
\dq test.%
\dt test.%
\dt test.t1
\dt+ test.t1
\dv test.%
\dv test.v1
\dv+ test.v1
\xfoox
\lt test.%;
\lt test.% ;
\q
SELECT 'should not get here';
