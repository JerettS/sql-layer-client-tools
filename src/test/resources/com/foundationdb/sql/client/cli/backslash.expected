ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

ROWS: 0

 Schema | Name | Type 
--------+------+------
(0 rows)

 Schema | Name |   Type   
--------+------+----------
 test   | a    | TABLE    
 test   | c    | TABLE    
 test   | i    | TABLE    
 test   | o    | TABLE    
 test   | s1   | SEQUENCE 
 test   | s2   | SEQUENCE 
 test   | t1   | TABLE    
 test   | t2   | TABLE    
 test   | v1   | VIEW     
 test   | v2   | VIEW     
(10 rows)

 Schema | Name |   Type   
--------+------+----------
 test   | s1   | SEQUENCE 
 test   | t1   | TABLE    
 test   | v1   | VIEW     
(3 rows)

 Schema | Name | Type  
--------+------+-------
 test   | t1   | TABLE 
(1 row)

 Schema | Name |   Type   
--------+------+----------
 test   | s1   | SEQUENCE 
(1 row)

 Schema | Name | Type 
--------+------+------
 test   | v1   | VIEW 
(1 row)

 Schema | Table |    Name    
--------+-------+------------
 test   | a     | PRIMARY    
 test   | a     | aid        
 test   | c     | PRIMARY    
 test   | i     | PRIMARY    
 test   | i     | ofk        
 test   | o     | PRIMARY    
 test   | o     | cfk        
 test   | o     | odate_name 
 test   | t1    | i1         
 test   | t2    | i2         
(10 rows)

 Schema | Table | Name 
--------+-------+------
 test   | t1    | i1   
(1 row)

 Schema | Table | Name 
--------+-------+------
 test   | t1    | i1   
(1 row)

 Schema | Name 
--------+------
 test   | s1   
 test   | s2   
(2 rows)

 Schema | Name 
--------+------
 test   | s1   
(1 row)

 Schema | Name 
--------+------
 test   | a    
 test   | c    
 test   | i    
 test   | o    
 test   | t1   
 test   | t2   
(6 rows)

 Schema | Name 
--------+------
 test   | t1   
(1 row)

 Schema | Name 
--------+------
 test   | v1   
 test   | v2   
(2 rows)

 Schema | Name 
--------+------
 test   | v1   
(1 row)

 Table test.a
 Column | Type | Nullable 
--------+------+----------
 cid    | INT  | NO       
 aid    | INT  | NO       
(2 rows)

Indexes:
    PRIMARY (cid, aid)
    UNIQUE aid (aid)
References:
    GROUPING FOREIGN KEY (cid) REFERENCES c (cid)

 Table test.c
 Column |   Type   | Nullable 
--------+----------+----------
 cid    | INT      | NO       
 name   | CHAR(32) | YES      
(2 rows)

Indexes:
    PRIMARY (cid)
Referenced By:
    TABLE o CONSTRAINT o.cfk FOREIGN KEY (cid) REFERENCES c (cid)
    TABLE a GROUPING FOREIGN KEY (cid) REFERENCES c (cid)
    TABLE o GROUPING FOREIGN KEY (cid) REFERENCES c (cid)

 Table test.i
 Column | Type | Nullable 
--------+------+----------
 iid    | INT  | NO       
 oid    | INT  | YES      
(2 rows)

Indexes:
    PRIMARY (iid)
    ofk (oid)
References:
    i.ofk FOREIGN KEY (oid) REFERENCES o (oid)
    GROUPING FOREIGN KEY (oid) REFERENCES o (oid)

 Table test.o
 Column | Type | Nullable 
--------+------+----------
 oid    | INT  | NO       
 odate  | DATE | YES      
 cid    | INT  | YES      
(3 rows)

Indexes:
    PRIMARY (oid)
    cfk (cid)
    odate_name (o.odate, c.name) USING LEFT JOIN
References:
    o.cfk FOREIGN KEY (cid) REFERENCES c (cid)
    GROUPING FOREIGN KEY (cid) REFERENCES c (cid)
Referenced By:
    TABLE i CONSTRAINT i.ofk FOREIGN KEY (oid) REFERENCES o (oid)
    TABLE i GROUPING FOREIGN KEY (oid) REFERENCES o (oid)

 Sequence test.s1
  Type  | Start | Min |         Max         | Inc | Cycle 
--------+-------+-----+---------------------+-----+-------
 bigint |     1 |   1 | 9223372036854775807 |   1 | NO    
(1 row)

 Sequence test.s2
  Type  | Start | Min |         Max         | Inc | Cycle 
--------+-------+-----+---------------------+-----+-------
 bigint |     1 |   1 | 9223372036854775807 |   1 | NO    
(1 row)

 Table test.t1
 Column |    Type     | Nullable 
--------+-------------+----------
 c1     | INT         | YES      
 c2     | VARCHAR(32) | YES      
(2 rows)

Indexes:
    i1 (c1)

 Table test.t2
 Column |      Type      | Nullable 
--------+----------------+----------
 c1     | DECIMAL(10, 5) | YES      
 c2     | VARBINARY(10)  | YES      
(2 rows)

Indexes:
    i2 (c1)

 View test.v1
 Column |    Type     | Nullable 
--------+-------------+----------
 c1     | INT         | YES      
 c2     | VARCHAR(32) | YES      
(2 rows)

 View test.v2
 Column |      Type      | Nullable 
--------+----------------+----------
 c1     | DECIMAL(10, 5) | YES      
 c2     | VARBINARY(10)  | YES      
(2 rows)

 Sequence test.s1
  Type  | Start | Min |         Max         | Inc | Cycle 
--------+-------+-----+---------------------+-----+-------
 bigint |     1 |   1 | 9223372036854775807 |   1 | NO    
(1 row)

 Sequence test.s2
  Type  | Start | Min |         Max         | Inc | Cycle 
--------+-------+-----+---------------------+-----+-------
 bigint |     1 |   1 | 9223372036854775807 |   1 | NO    
(1 row)

 Table test.a
 Column | Type | Nullable 
--------+------+----------
 cid    | INT  | NO       
 aid    | INT  | NO       
(2 rows)

Indexes:
    PRIMARY (cid, aid)
    UNIQUE aid (aid)
References:
    GROUPING FOREIGN KEY (cid) REFERENCES c (cid)

 Table test.c
 Column |   Type   | Nullable 
--------+----------+----------
 cid    | INT      | NO       
 name   | CHAR(32) | YES      
(2 rows)

Indexes:
    PRIMARY (cid)
Referenced By:
    TABLE o CONSTRAINT o.cfk FOREIGN KEY (cid) REFERENCES c (cid)
    TABLE a GROUPING FOREIGN KEY (cid) REFERENCES c (cid)
    TABLE o GROUPING FOREIGN KEY (cid) REFERENCES c (cid)

 Table test.i
 Column | Type | Nullable 
--------+------+----------
 iid    | INT  | NO       
 oid    | INT  | YES      
(2 rows)

Indexes:
    PRIMARY (iid)
    ofk (oid)
References:
    i.ofk FOREIGN KEY (oid) REFERENCES o (oid)
    GROUPING FOREIGN KEY (oid) REFERENCES o (oid)

 Table test.o
 Column | Type | Nullable 
--------+------+----------
 oid    | INT  | NO       
 odate  | DATE | YES      
 cid    | INT  | YES      
(3 rows)

Indexes:
    PRIMARY (oid)
    cfk (cid)
    odate_name (o.odate, c.name) USING LEFT JOIN
References:
    o.cfk FOREIGN KEY (cid) REFERENCES c (cid)
    GROUPING FOREIGN KEY (cid) REFERENCES c (cid)
Referenced By:
    TABLE i CONSTRAINT i.ofk FOREIGN KEY (oid) REFERENCES o (oid)
    TABLE i GROUPING FOREIGN KEY (oid) REFERENCES o (oid)

 Table test.t1
 Column |    Type     | Nullable 
--------+-------------+----------
 c1     | INT         | YES      
 c2     | VARCHAR(32) | YES      
(2 rows)

Indexes:
    i1 (c1)

 Table test.t2
 Column |      Type      | Nullable 
--------+----------------+----------
 c1     | DECIMAL(10, 5) | YES      
 c2     | VARBINARY(10)  | YES      
(2 rows)

Indexes:
    i2 (c1)

 Table test.t1
 Column |    Type     | Nullable 
--------+-------------+----------
 c1     | INT         | YES      
 c2     | VARCHAR(32) | YES      
(2 rows)

Indexes:
    i1 (c1)

 Table test.t1
 Column |    Type     | Nullable | Default | Charset | Collation | Sequence 
--------+-------------+----------+---------+---------+-----------+----------
 c1     | INT         | YES      | 10      |         |           |          
 c2     | VARCHAR(32) | YES      |         | utf8    | en_ci     |          
(2 rows)

Indexes:
    i1 (c1)

 View test.v1
 Column |    Type     | Nullable 
--------+-------------+----------
 c1     | INT         | YES      
 c2     | VARCHAR(32) | YES      
(2 rows)

 View test.v2
 Column |      Type      | Nullable 
--------+----------------+----------
 c1     | DECIMAL(10, 5) | YES      
 c2     | VARBINARY(10)  | YES      
(2 rows)

 View test.v1
 Column |    Type     | Nullable 
--------+-------------+----------
 c1     | INT         | YES      
 c2     | VARCHAR(32) | YES      
(2 rows)

 View test.v1
 Column |    Type     | Nullable | Default | Charset | Collation | Sequence 
--------+-------------+----------+---------+---------+-----------+----------
 c1     | INT         | YES      |         |         |           |          
 c2     | VARCHAR(32) | YES      |         | utf8    | en_ci     |          
(2 rows)

Invalid command: \xfoox. Try \? for help.

 Schema | Name 
--------+------
 test   | a    
 test   | c    
 test   | i    
 test   | o    
 test   | t1   
 test   | t2   
(6 rows)

 Schema | Name 
--------+------
 test   | a    
 test   | c    
 test   | i    
 test   | o    
 test   | t1   
 test   | t2   
(6 rows)
